src/script/sigcache.h

```diff
@@ -71,6 +72,22 @@ class CachingTransactionSignatureChecker : public TransactionSignatureChecker
 
     bool VerifyECDSASignature(const std::vector<unsigned char>& vchSig, const CPubKey& vchPubKey, const uint256& sighash) const override;
     bool VerifySchnorrSignature(Span<const unsigned char> sig, const XOnlyPubKey& pubkey, const uint256& sighash) const override;
+
+    bool GetStore() const { return store; }
+    SignatureCache& GetSigCache() const { return m_signature_cache; }
+};
+
+class CollectingSignatureChecker : public CachingTransactionSignatureChecker {
```

Comment:
https://github.com/bitcoin/bitcoin/pull/29491/commits/d333d5c6e3c4b454621998772ff74ff6656af691: This is nice! sets us up nicely to later do batch taptweak checking since we can "collect" it too.